### DISCUSSION TITLE
the output of edge_updater for undirected graph

### DISCUSSION THREAD

**Discussion by Powerd0g** [pyg-team/pytorch_geometric#4633] [Q&A]
I use the `edge_updater` to update edge embedding, but for undirected graph, it will return twice the number of edges.
if the `edge_updater` could return the same number as the edge?
for example 
edge_index:
[[0, 0, 0, 1, 2, 3]
[1, 2, 3, 0, 0, 0]]
the `edge_updater` will return edge 0-1, 0-2, 0-3 and 1-0, 2-0, 3-0
but i only want the edge 0-1, 0-2, 0-3

**Comment by rusty1s** [comment]
If you only want a single direction, you can do:
```python
mask = edge_index[0] <= edge_index[1]

self.edge_updater(edge_index[:, mask], ...)
```
